Improve feedback guidance (#160)
canonical/ubuntu-maintainers-handbook: MergeProposalReview.md

has helped us maintain consistently high quality.
To do so we propose our changes as [merge proposals against `git-ubuntu` branches](MergeProposal.md).

## Review Process

In this review, we:
* Review formal content for issues
* Ensure adherence to team style guidelines
* Identify common issues
* Conduct further testing (when warranted)

> ### Chose the appropriate level of scrutiny

Author: applied

Before we go into any details on our review process we have to admit that,
sometimes a proposal can be rather trivial.
And while we love the rigor we have established and holding each other
accountable to it, we also hate inneficiency.

> Therefore we acknoledge that we have two kind of changes we might propose:

Reviewer: s/acknoledge/acknowledge as well

* Changes are complex or there is any kind of remaining uncertainty
  * Full review please as outlined below (the default)
* Changes are trivial:
  * The requester is only looking for a second pair of eyes for lapses that might have happened unaware and unintentionally. But not after a deper check as we'd usually do.
  * In this case the requester shall mark the proposal by stating "Trivial change" in the description.

>   * Depending on the case that might also lift some other common practise we have like pre-building in PPA and pre-testing autopkgtest against it

Reviewer: What do you mean by "lift"? Is this something understood by the team as a whole (that I just don't have the context on)?

Author: @s-makin changed it to be more clear "Depending on the case, trivial changes can also free from other common practice we have, like pre-building in PPA and pre-testing autopkgtest against it." 

Also moved it to the last bullet point (as it is different than the rest)

Also fixed another typo just above.

Author: even better
"Depending on the case such trivial changes can also free from further, otherwise required, practice that we have. Like pre-building in PPA and pre-testing autopkgtest against it."

  * In this case the review can be free form and rather short
  * But if the reviewer has a hunch or disagrees with that classification they are entitled to do a full pass.

### States for each template entry

The following states indicate reviewer feedback on each aspect we check.

* *OK* - This was checked and there was nothing of concern to be raised.
* *Not OK* - Something is not as it should be and needs to be changed in the proposed content.

> * *Question* - Not neccesarily bad, but a detail worth to clarify. This could only need an answer, but might due to the discussion also end up in changes to what was proposed.

Reviewer: ```suggestion
* *Question* - Not necessarily bad, but a detail worth clarifying. This could only need an answer, but might as a result of the discussion also end up in changes to what was proposed.
```

* *Not applicable* - The particular check is not applicable to this case
* *Skipped* - Skipped by the reviewer as it seemed not mandatory or commensurate for this case.

### Skipping whole sections

Not every change needs all the checks we have in our template. If you would end
up marking all elements of a section as `skipped` feel free to drop the whole
section to make the post have a lower percentage of noisy non-content text.

A common case for this are the checks in regard to "package merges", "new delta
in debian/", or "new patches".

### How to provide feedback details

On any of these states further context might be given in the line below. That is
entirely optional on "OK" or "Not applicable" and to some extent also for "Skipped" states.

> But any "Not OK" and "Question" state the reviewer shall provide some details

Reviewer 2: todo:
```suggestion
But for any "Not OK" and "Question" state the reviewer shall provide some details
```

to allow the requester to understand and act on it.

Furthermore when providing feedback we strive to follow the style of

> [conventional comments](https://conventionalcomments.org/) which allow us to

Reviewer: ```suggestion
[conventional comments](https://conventionalcomments.org/) which allows us to
```

bring up a lot - thereby helping each other - without locking us too much in
review iterations. Such a simple prefix helps in many common cases, making it
clear if things are blocking or optional.
Reviews do not even need to be only critique, if you see something nice

> remember the `praise:` label exists.

Reviewer 2: praise: I've been making a very good experience using "conventional comments" on my daily Netplan reviews in the past for a wide community of people. Especially the "(non-blocking)" decoration makes very clear something is optional.

Reviewer: question: do we want to adopt these in the documentation as well? I'm happy to add a reference to it in the contributing guide if so. Conventional commits didn't make any sense for a pure documentation repo, but this is a nice review-friendly version where we could align practice across the whole team.

Reviewer: Sounds good to me - I'll write it up as such and add it to our contributor docs

## What we review

The following is an overview of common things we look out for. Obviously each

> case is different and a review can also check for anything else they consider

Reviewer: ```suggestion
case is different and a reviewer can also check for anything else they consider
```

  * In some situations, manual verification of certain behaviors may also be necessary for additional assurance.

## Review template

The following template may be useful when submitting reviews, it covers all

> the usual checks we describe above making it an easy checkbox approach to not

Reviewer: I prefer the second alternative - it's a lot clearer imo.

> the usual checks we describe above making it an easy checkbox approach to not

Author: I prefer the second as well - applied
